### PR TITLE
Add PLL base address and extend clock helper functions

### PR DESCRIPTION
**Pull request by alex96295**

### CODE REVIEW

**Comment by alex96295** [comment]
Rebased onto main

File: sw/include/car_util.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ enum car_clk {
     CAR_SECURITY_CLK = 3,
     CAR_PULP_CLK     = 4,
     CAR_SPATZ_CLK    = 5,
-    CAR_L2_CLK       = 6,
```
Comment (bluewww) [inline sw/include/car_util.h:30]: this is C note SystemVerilog so it is totally valid and allows for symmetry

File: sw/include/car_util.h [hunk 2]
```diff
@@ -87,7 +83,7 @@ enum car_rst {
 	    return CARFIELD_SPATZ_CLUSTER_##NAME##_REG_OFFSET;                 \
 	case CAR_L2_RST:                                                       \
 	    return CARFIELD_L2_##NAME##_REG_OFFSET;                            \
-	default:                                                               \
```
Comment (bluewww) [inline sw/include/car_util.h:67]: ?

Comment (alex96295) [reply sw/include/car_util.h:67]: my bad

File: sw/include/car_util.h [hunk 3]
```diff
@@ -13,13 +13,36 @@
...
 #define SAFETY_ISLAND_BOOT_ADDR_RSVAL (CAR_SAFETY_ISLAND_PERIPHS_BASE_ADDR + 0x1080)
 
 enum car_isolation_status { CAR_ISOLATE_DISABLE = 0, CAR_ISOLATE_ENABLE = 1 };
 
 enum car_rst_status { CAR_RST_ASSERT = 1, CAR_RST_RELEASE = 0 };
 
+// As input clocks for carfield, we have 3 clock sources (`clk_srcs`) to be
+// multiplexed among 6 domains (`car_clk`).
+enum clk_src {
+    CLK0 = 0,
```
Comment (bluewww) [inline sw/include/car_util.h:41]: CAR_*

**Comment by alex96295** [comment]
@bluewww I decoupled these commits from #104 and this PR is ready for review

File: sw/include/car_util.h [hunk 4]
```diff
@@ -20,6 +20,13 @@ enum car_isolation_status { CAR_ISOLATE_DISABLE = 0, CAR_ISOLATE_ENABLE = 1 };
 
 enum car_rst_status { CAR_RST_ASSERT = 1, CAR_RST_RELEASE = 0 };
 
+// As input clocks for carfield, we have 3 clock sources to be multiplexed among 6 domains.
+enum clk_src {
```
Comment (bluewww) [inline sw/include/car_util.h:24]: car_src_clk

File: sw/include/car_util.h [hunk 5]
```diff
@@ -30,7 +37,7 @@ enum car_clk {
     CAR_L2_CLK       = 6,
 };
 
-enum car_rst {
+enum car_domain_rst {
```
Comment (bluewww) [inline sw/include/car_util.h:40]: should be `car_rst` please revert

File: sw/include/car_util.h [hunk 6]
```diff
@@ -20,7 +20,14 @@ enum car_isolation_status { CAR_ISOLATE_DISABLE = 0, CAR_ISOLATE_ENABLE = 1 };
...
 
-enum car_clk {
+// As input clocks for carfield, we have 3 clock sources to be multiplexed among 6 domains.
+enum car_src_clk {
+    CAR_CLK0 = 0,
+    CAR_CLK1 = 1,
+    CAR_CLK2 = 2
+};
+
+enum car_domain_clk {
```
Comment (bluewww) [inline sw/include/car_util.h:30]: Should be car_clk please revert.

This is about the clock signals not the clock domains. Clocks signals also don't correspond to clock domains necessarily

Comment (alex96295) [reply sw/include/car_util.h:30]: I do not agree, `car_src_clk` is about clock signals, `*_domain_*` is referred to the clk or the rst of a domain, and in the enum you have indeed the domains (clk or rst according to the enum type)

Comment (bluewww) [reply sw/include/car_util.h:30]: The enums are about the clocks and the reset signals. I used to call it `domain` but then I changed it

Comment (bluewww) [reply sw/include/car_util.h:30]: Any other reference to domain is probably because I didn't update it correctly

Comment (alex96295) [reply sw/include/car_util.h:30]: mmh no there were not leftovers of your past naming scheme, I just thought to update the names with `domain` after adding the `car_src_clk` enum; anyway, let's keep it short I can revert it

**Comment by alex96295** [comment]
Merging as soon as CI passes

**Review by micprog — APPROVED** [review]
LGTM 👍

**Pull request merged**